### PR TITLE
8331733: [PPC64] saving and restoring CR is not needed at most places

### PR DESCRIPTION
**Pull request by varada1110**
PPC64 uses save/restore CR less often. Only LR is critical, CR is mainly needed for native-to-Java calls.
Fastdebug: build and tier1 testing successful. [unrelated failures]

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8331733](https://bugs.openjdk.org/browse/JDK-8331733): [PPC64] saving and restoring CR is not needed at most places (**Enhancement** - P4)

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Amit Kumar](https://openjdk.org/census#amitkumar) (@offamitkumar - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19494/head:pull/19494` \
`$ git checkout pull/19494`

Update a local copy of the PR: \
`$ git checkout pull/19494` \
`$ git pull https://git.openjdk.org/jdk.git pull/19494/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19494`

View PR using the GUI difftool: \
`$ git pr show -t 19494`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19494.diff">https://git.openjdk.org/jdk/pull/19494.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19494#issuecomment-2147030317)

### CODE REVIEW

File: src/hotspot/cpu/ppc/gc/shared/barrierSetAssembler_ppc.cpp [hunk 1]
```diff
@@ -342,7 +342,7 @@ int SaveLiveRegisters::iterate_over_register_mask(IterationAction action, int of
         }
       }
     } else if (vm_reg->is_ConditionRegister()) {
-      // NOP. Conditions registers are covered by save_LR_CR
+      // NOP. Conditions registers are covered by save_LR
```
Comment (TheRealMDoerr) [inline src/hotspot/cpu/ppc/gc/shared/barrierSetAssembler_ppc.cpp:345]: This comment is no longer correct. I don't think that we ever save or restore condition registers at this point. So, I think we can replace this comment by `ShouldNotReachHere(); // live condition registers are unexpected at this point`.

Comment (varada1110) [reply src/hotspot/cpu/ppc/gc/shared/barrierSetAssembler_ppc.cpp]: Thanks @TheRealMDoerr . I have changed the comment

**Comment by TheRealMDoerr** [comment]
Please make xBarrierSetAssembler_ppc.cpp consistent with the normal BarrierSetAssembler. "// NOP. Conditions registers are covered by save_LR_CR" should get replaced, too.
Saving and restoring CR in `RegisterSaver::push_frame_reg_args_and_save_live_registers` and `RegisterSaver::restore_live_registers_and_pop_frame` are not needed, either.
Otherwise, this PR looks good to me. I'll run tests. (Will retest when you make updates.)

**Comment by varada1110** [comment]
Thank you @TheRealMDoerr . I have made the changes.

**Comment by TheRealMDoerr** [comment]
Thanks for the updates! Sorry, my assumption is wrong. I have found a test which runs into the `ShouldNotReachHere()`: jdk/incubator/vector/VectorMaxConversionTests.java#ZGenerational
I think we should revert all changes in `SaveLiveRegisters` (barrierSetAssembler_ppc.cpp) and `XSaveLiveRegisters` (xBarrierSetAssembler_ppc.cpp). We can keep them saving and restoring CR. Sorry for not finding this earlier.

**Comment by varada1110** [comment]
> Thanks for the updates! Sorry, my assumption is wrong. I have found a test which runs into the `ShouldNotReachHere()`: jdk/incubator/vector/VectorMaxConversionTests.java#ZGenerational I think we should revert all changes in `SaveLiveRegisters` (barrierSetAssembler_ppc.cpp) and `XSaveLiveRegisters` (xBarrierSetAssembler_ppc.cpp). We can keep them saving and restoring CR. Sorry for not finding this earlier.

No problem @TheRealMDoerr . Thanks for fix. I have reverted the changes and now the test jdk/incubator/vector/VectorMaxConversionTests.java is passing. I will run tier1 one more time

**Comment by TheRealMDoerr** [comment]
Please also revert the two ShouldNotReachHere changes. The old comments are correct again, now. Note that VectorMaxConversionTests.java is in tier3.

**Review by TheRealMDoerr — COMMENTED** [review]
Looks good, now. Thanks. I'll rerun the tests.

**Review by offamitkumar — CHANGES_REQUESTED** [review]
Please update the copyright headers as well.

**Comment by TheRealMDoerr** [comment]
The tests have passed on our side. I can approve it once the Copyright years are updated.

**Comment by varada1110** [comment]
Thanks @TheRealMDoerr for running the tests. Only tier1 is completed from my side which looks fine. @offamitkumar I have updated the copyright headers. Thank you

**Review by offamitkumar — APPROVED** [review]
Looks good to me. 

I ran tier1 & vector test on `ppc-le` and these are the results: 
```
failures in tier1 (looks unrelated to this PR): 
gtest/GTestWrapper.java 
java/util/ResourceBundle/Control/MissingResourceCauseTestRun.java 
jdk/javadoc/doclet/testIOException/TestIOException.java 

````

```
Vector tests: 

Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR   
   jtreg:./test/jdk/jdk/incubator/vector                78    78     0     0   
==============================
TEST SUCCESS
``

**Review by TheRealMDoerr — APPROVED** [review]

**Comment by varada1110** [comment]
Thank you,
/integrate

**Pull request closed without merging**